### PR TITLE
IGNITE-18850 Add ssl configuration validation

### PR DESCRIPTION
**Pull request by valepakh**
https://issues.apache.org/jira/browse/IGNITE-18850

### CODE REVIEW

File: modules/network/src/test/java/org/apache/ignite/internal/network/configuration/SslConfigurationValidatorImplTest.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    @Test
+    public void nullTrustStorePath(@TempDir Path tmpDir) throws IOException {
+        validate(createTrustStoreConfig(tmpDir, "PKCS12", null, "changeIt"),
+                "Key store path must not be blank");
+    }
+
+    @Test
+    public void emptyTrustStorePath(@TempDir Path tmpDir) throws IOException {
+        validate(createTrustStoreConfig(tmpDir, "PKCS12", "", "changeIt"),
+                "Key store path must not be blank");
```
Comment (PakhomovAlexander) [inline modules/network/src/test/java/org/apache/ignite/internal/network/configuration/SslConfigurationValidatorImplTest.java:65]: I think it is better to print trust store in those cases, not key store.

**Review by PakhomovAlexander — COMMENTED** [review]
Have you tested the patch manually or with integration tests?

**Comment by valepakh** [comment]
> Have you tested the patch manually or with integration tests?

Not sure what do you mean, integration tests were run during the check.

**Review by PakhomovAlexander — APPROVED** [review]
LGTM

File: modules/network/src/main/java/org/apache/ignite/internal/network/configuration/SslConfigurationValidatorImpl.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import org.apache.ignite.configuration.validation.ValidationContext;
+import org.apache.ignite.configuration.validation.ValidationIssue;
+import org.apache.ignite.configuration.validation.Validator;
+
+/**
+ * SSL configuration validator implementation.
+ */
+public class SslConfigurationValidatorImpl implements Validator<SslConfigurationValidator, SslView> {
+
+    public static final SslConfigurationValidatorImpl INSTANCE = new SslConfigurationValidatorImpl();
```
Comment (ibessonov) [inline modules/network/src/main/java/org/apache/ignite/internal/network/configuration/SslConfigurationValidatorImpl.java:34]: I never understood, why do we need a static instance. I understand that you just replicated the approach, but nonetheless, I'm curious

File: modules/network/src/main/java/org/apache/ignite/internal/network/configuration/SslConfigurationValidatorImpl.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+    @Override
+    public void validate(SslConfigurationValidator annotation, ValidationContext<SslView> ctx) {
+        SslView ssl = ctx.getNewValue();
+        if (ssl.enabled()) {
+            validateKeyStore(ctx, ".keyStore", "Key store", ssl.keyStore());
+
+            ClientAuth clientAuth = ClientAuth.valueOf(ssl.clientAuth().toUpperCase());
```
Comment (ibessonov) [inline modules/network/src/main/java/org/apache/ignite/internal/network/configuration/SslConfigurationValidatorImpl.java:42]: I think it'll throw an exception if enum element is not found, is that the expected behavior?

File: modules/network/src/main/java/org/apache/ignite/internal/network/configuration/SslConfigurationValidator.java [hunk 4]
```diff
@@ -23,9 +23,9 @@
 import java.lang.annotation.Target;
 
 /**
- * Annotation to validate whole key store configuration.
+ * Annotation to validate SSL configuration.
  */
 @Target(ElementType.FIELD)
 @Retention(RetentionPolicy.RUNTIME)
-public @interface KeyStoreConfigurationValidator {
+public @interface SslConfigurationValidator {
```
Comment (ibessonov) [inline modules/network/src/main/java/org/apache/ignite/internal/network/configuration/SslConfigurationValidator.java:30]: I feel like the annotation is unused

File: modules/network/src/main/java/org/apache/ignite/internal/network/configuration/SslConfigurationValidatorImpl.java (hunk 3 above)
Comment (valepakh) [reply modules/network/src/main/java/org/apache/ignite/internal/network/configuration/SslConfigurationValidatorImpl.java]: I added the check for this and also for the possible exception when checking the path.

File: modules/network/src/main/java/org/apache/ignite/internal/network/configuration/SslConfigurationValidator.java (hunk 4 above)
Comment (valepakh) [reply modules/network/src/main/java/org/apache/ignite/internal/network/configuration/SslConfigurationValidator.java:30]: Nice catch, fixed!

**Review by ibessonov — APPROVED** [review]

**Pull request merged**